--- a/crypto/cversion.c
+++ b/crypto/cversion.c
@@ -71,7 +71,7 @@
 # ifdef OPENSSL_USE_BUILD_DATE
         return (DATE);
 # else
-        return ("built on: reproducible build, date unspecified");
+        return ("built on: reproducible build, date suppressed");
 # endif
 #else
         return ("built on: date not available");
@@ -79,7 +79,11 @@
     }
     if (t == SSLEAY_CFLAGS) {
 #ifdef CFLAGS
+# ifdef OPENSSL_USE_BUILD_DATE
         return (CFLAGS);
+# else
+        return ("compiler: reproducible build, cflags suppressed");
+# endif
 #else
         return ("compiler: information not available");
 #endif
